src/app/core/utils/ComponentOrganizers.ts

```diff
@@ -0,0 +1,55 @@
+import {Vector} from "Vector";
+import {Graph} from "math/Graph";
+
+import {CreateGraph, IOObjectSet} from "./ComponentUtils";
+
+import {Component} from "core/models/Component";
+
+function OrganizeCore(groups: IOObjectSet, start: Vector, depthMap: Map<number, number>): void {
+    const depthArray: Component[][] = [];
+    const components = groups.getComponents();
+
+    // Convert the map to a 2d array where index of first array is the depth
+    for (const [index, depth] of depthMap.entries()) {
+        while (depthArray.length <= depth)
+            depthArray.push([]);
+        depthArray[depth].push(components[index]);
+    }
+
+    const xOffset: number = 250;
+    const yOffset: number = 150;
+
+    for (let i = 0; i < depthArray.length; i++) {
+        for (let j = 0; j < depthArray[i].length; j++) {
+            const component = depthArray[i][j];
+            const newLocation = new Vector(start.x + xOffset * i, start.y + yOffset * j);
+            component.setPos(newLocation);
+        }
+    }
+}
```

Comment:
I agree this whole thing is incredibly convoluted, should I try work on refactoring it or is there some other approach that I should check out instead?

Reply:
Okay, so I took a closer look and I believe I know what you're trying to do.
Basically, you created this Map of Nodes to their Depth, and then in this method, you are basically finding the inverse, i.e., the map of each Depth to every Node at that depth, and then using that information to place it.

I believe the best course of action here is to actually take in a `Map<number, number[]>` and changed `Graph.getNodeDepths` to find the Mapping of depths to nodes instead of nodes to depths.

This would drastically simplify the OrganizeCore method to something like:
```ts
function OrganizeCore(groups: IOObjectSet, start: Vector, depthMap: Map<number, number[]>): void {
    // Loop through each depth-level
    Array.from(depthMap.entries()).forEach(([depth, indices]) =>
        // Then loop through each component at each depth and set position
        //  with x-position related to `depth` and y-position related to `i`
        indices.map(c => groups.getComponents()[c])
               .forEach((c, i) => c.setPos(start.add(ORGANIZE_SEP_X*depth - depthMap.size/2, 
                                                     ORGANIZE_SEP_Y*i - indices.length/2)))
    );
}
```

Also, for the constants for `offset` you should instead make them constants in `core/utils/Constants` and call them something like I named them above (`ORGANIZE_SEP_X`, `ORGANIZE_SEP_Y`)


Also for below for `OrganizeMax/MinDepth`
you can 1-line them and have it just be `OrganizeCore(groups, start, CreateGraph(groups).getMaxNodeDepths())`

Reply:
Made these changes, but rather than have depthMap be a Map, I just used a 2d array where each "inner array" contains the components for a given depth, and the index of that array is the the depth that those components are at.